data/migrations/versions/5b8dc452f5c3_add_tag_immutability_column_and_logentry_kinds.py

```diff
@@ -0,0 +1,70 @@
+"""add tag immutability column
+
+Revision ID: 5b8dc452f5c3
+Revises: 41d15c93c299
+Create Date: 2023-05-21 13:13:10.565161
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "5b8dc452f5c3"
+down_revision = "41d15c93c299"
+
+import sqlalchemy as sa
+
+
+def upgrade(op, tables, tester):
+    # add column to track tack immutability
+    op.add_column(
+        table_name="tag",
+        column=sa.Column(
+            name="immutable",
+            type_=sa.Boolean(),
+            nullable=False,
+            server_default=sa.sql.expression.false(),
+        ),
+    )
```

Comment:
I am not sure about other databases, but Postgres should be quite fast if I understand correctly (since it's a non-volatile default value).  Given that we deprecated MySQL for a while now, the advantage of avoiding the overhead of constantly doing joins and lookups for the majority of the users outweighs a slow migration / Quay update for a very small number of users.
On the other hand, how heavy would the overhead be if a join from a relatively small and likely completely cached table constantly calculates the tag immutability info?

Reply:
if we resorted to using the joins the overhead would be low but ideally we could just add the column and prevent the database structure from becoming fragmented into different tables. To do that IMO we need some sort of confirmation that when the migration runs on Quay.io it doesn't take more resources than expected and impact the service. I'm open to other opinions @syed @kleesc

Reply:
My hope is that adding a column that has a static default value is almost a no-op for Postgres and it's MVCC, because the value can be computed when read and doesn't have to be written to a new row internally unless actually set to `True` or due to another update of the row (e.g. expiry date)